src/server/plugins/engine/models/FormModel.ts

```diff
@@ -284,11 +284,16 @@ export class FormModel {
     const { isPreview } = checkFormStatus(request.path)
 
     // Add paths for navigation
-    for (const { collection, path } of context.relevantPages) {
+    for (const { keys, path } of context.relevantPages) {
       paths.push(path)
 
-      // Stop at current page or with errors
-      if (!isPreview && collection.getErrors(errors)) {
+      // Stop at page with errors
+      if (
+        !isPreview &&
+        errors?.some(({ name, path }) => {
+          return keys.includes(name) || keys.some((key) => path.includes(key))
```

Comment:
Our helper `collection.getErrors()` only included errors from collection fields

State schema errors for the repeater `options.name` nested value were excluded:

https://github.com/DEFRA/forms-runner/blob/c1fecf5f43ffd233a9618005c80eb4d9b7a959c7/src/server/plugins/engine/pageControllers/RepeatPageController.ts#L44-L52